File: ClearScript/Windows/Core/JScriptEngine.InitScript.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license.





namespace Microsoft.ClearScript.Windows.Core
{
    public partial class JScriptEngine
    {
        internal const string InitScript = "EngineInternal=function(){var globalObject=this;function convertArgs(t){var n=[];if(t.GetValue)for(var o=t.Length,e=0;e<o;e++)n.push(t[e]);else for(var o=(t=new VBArray(t)).ubound(1)+1,e=0;e<o;e++)n.push(t.getItem(e));return n}function construct(t,n,o,e,r,u,i,f,c,p,s,a,h,l,y,v){return new this(t,n,o,e,r,u,i,f,c,p,s,a,h,l,y,v)}var savedJSON=globalObject.JSON;return{getCommandResult:function(t){return null!==t&&('object'==typeof t||'function'==typeof t)&&'function'==typeof t.toString?t.toString():t},invokeConstructor:function(t,n){if('function'!=typeof t)throw Error('Function expected');return construct.apply(t,convertArgs(n))},invokeMethod:function(t,n,o){if('function'!=typeof n)throw Error('Function expected');return n.apply(t,convertArgs(o))},isPromise:function(t){return!1},isHostObject:function(t){return!!t&&'function'!=typeof t.constructor&&t!==globalObject},throwValue:function(t){throw t},parseJson:function(json){return savedJSON?savedJSON.parse(json):eval('('+json+')')}}}();";
    }
}

File: ClearScript/Windows/Core/JScriptEngine.InitScript.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license.

EngineInternal = (function () {

    var globalObject = this;

    function convertArgs(args) {
        var result = [];
        if (args.GetValue) {
            var count = args.Length;
            for (var i = 0; i < count; i++) {
                result.push(args[i]);
            }
        }
        else {
            args = new VBArray(args);
            var count = args.ubound(1) + 1;
            for (var i = 0; i < count; i++) {
                result.push(args.getItem(i));
            }
        }
        return result;
    }

    function construct(arg0, arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8, arg9, arg10, arg11, arg12, arg13, arg14, arg15) {
        return new this(arg0, arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8, arg9, arg10, arg11, arg12, arg13, arg14, arg15);
    }

    var savedJSON = globalObject.JSON;

    return {

        getCommandResult: function (value) {
            if (value !== null) {
                if ((typeof(value) === 'object') || (typeof(value) === 'function')) {
                    if (typeof(value.toString) === 'function') {
                        return value.toString();
                    }
                }
            }
            return value;
        },

        invokeConstructor: function (constructor, args) {
            if (typeof(constructor) !== 'function') {
                throw new Error('Function expected');
            }
            return construct.apply(constructor, convertArgs(args));
        },

        invokeMethod: function (target, method, args) {
            if (typeof(method) !== 'function') {
                throw new Error('Function expected');
            }
            return method.apply(target, convertArgs(args));
        },

        isPromise: function (value) {
            return false;
        },

        isHostObject: function (value) {
            return !!value && (typeof(value.constructor) !== 'function') && (value !== globalObject);
        },

        throwValue: function (value) {
            throw value;
        },

        parseJson: function (json) {
            return savedJSON ? savedJSON.parse(json) : eval('(' + json + ')');
        }
    };
})();
